pysisyphus fails to build with nixpkgs-unstable

pysisyphus started to show build failures, with the latest version of nixpkgs-unstable (e.g. with pinning turned off: `nix-build release.nix --arg pin false -A qchem.pysisyphus`)

```
==================================== ERRORS ====================================
____________ ERROR collecting pysisyphus/tests/test_calculators.py _____________
ImportError while importing test module '/build/source/pysisyphus/tests/test_calculators.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/nix/store/k4rnxn7l5i1jzj33062wlw6g5rlvaydf-python3.9-xtb-python-20.2/lib/python3.9/site-packages/xtb/libxtb.py:32: in <module>
    from ._libxtb import ffi, lib
E   ModuleNotFoundError: No module named 'xtb._libxtb'

During handling of the above exception, another exception occurred:
/nix/store/i6vabb4div9iy6lsl642d86k1q8riasn-python3-3.9.9/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
pysisyphus/tests/test_calculators.py:7: in <module>
    from pysisyphus.calculators import (
pysisyphus/calculators/__init__.py:49: in <module>
    from pysisyphus.calculators.ONIOMv2 import ONIOM
pysisyphus/calculators/ONIOMv2.py:32: in <module>
    from pysisyphus.calculators import (
pysisyphus/calculators/PyXTB.py:3: in <module>
    from xtb.interface import Environment, Param, Calculator as XTBCalculator
/nix/store/k4rnxn7l5i1jzj33062wlw6g5rlvaydf-python3.9-xtb-python-20.2/lib/python3.9/site-packages/xtb/interface.py:24: in <module>
    from .libxtb import (
/nix/store/k4rnxn7l5i1jzj33062wlw6g5rlvaydf-python3.9-xtb-python-20.2/lib/python3.9/site-packages/xtb/libxtb.py:34: in <module>
    raise ImportError("xtb C extension unimportable, cannot use C-API")
E   ImportError: xtb C extension unimportable, cannot use C-API
```

CC @sheepforce

--- a/pkgs/lib/xtb-python/default.nix
+++ b/pkgs/lib/xtb-python/default.nix
@@ -1,5 +1,5 @@
 { buildPythonPackage, lib, fetchFromGitHub, cffi, numpy, ase, qcelemental, meson, ninja, pkg-config,
-  xtb, pytest
+  xtb, pytestCheckHook, python
 }:
 
 buildPythonPackage rec {
@@ -33,13 +33,14 @@ buildPythonPackage rec {
 
   # Build a C module to interface XTB.
   preBuild = ''
-    meson setup build --prefix=$PWD --default-library=shared
+    meson setup build --prefix=$(pwd) --default-library=shared
     ninja -C build install
+    cp ./${python.passthru.sitePackages}/_libxtb.*.so ./xtb/.
   '';
 
-  checkInputs = [ pytest ];
-  checkPhase = "pytest";
-  doCheck = false;
+  checkInputs = [ pytestCheckHook ];
+  pytestFlagsArray = [ "-k 'not qcschema'" ]; # Numerically soooo slightly off
+  pythonImportsCheck = [ "xtb.interface" "xtb.libxtb" ];
 
   meta = with lib; {
     description = "Python wrapper for the semiempirical XTB package";

